src/btreemap.rs

```diff
@@ -60,6 +58,9 @@ struct BTreeHeader {
     _buffer: [u8; 24],
 }
 
+/// Current header size with additional header space
+const HEADER_SIZE: usize = 52;
```

Comment:
Now that we have this constant, I think it'd useful to document the header layout just like we do [here](https://github.com/dfinity/stable-structures/blob/main/src/vec.rs#L3). WDYT?